lib/live-collection.js

```diff
@@ -191,10 +194,27 @@ module.exports = Backbone.Collection.extend({
       this.reset(snapshot, options);
     }
 
+    this.snapshotReceived = true;
+
     this.trigger('sync');
     this.trigger('snapshot');
   },
 
+  onSnapshotReceived: function() {
+    if (this.snapshotReceived) {
```

Comment:
This implementation currently doesn't take into account the collection being reset after a Bayeux reset. 

Putting a `_snapshotReceived = false` on the first line of `_resetOptional` (before the shortcut) is probably the best place to put this.